internal/martian/proxy.go

```diff
@@ -836,7 +850,19 @@ func (p *Proxy) connectHTTP(req *http.Request, proxyURL *url.URL) (res *http.Res
 	} else {
 		d = dialvia.HTTPProxy(p.dial, proxyURL)
 	}
-	d.ConnectRequestModifier = p.ConnectRequestModifier
+	d.ConnectRequestModifier = func(r *http.Request) error {
```

Comment:
Maybe we should have it as a function `p.connectRequestModifier` WDYT?

Reply:
The only problem is that we need to pass a context from the original request,  so it would be 
```go
p.connectRequestModifier(ctx context.Context)
```
but sure I can do that.